### PR TITLE
Use Additional Fields In ActionRules

### PR DESCRIPTION
**Pull request by jeberl**
## THIS IS A STCKED PR. Only review commit a25c0c0

The other commits are included in PR #742

Summary
---------

This diff enables users to add conditions to their action rules based on Additional Fields included on the content. A use case for this might be a different queue for mobile uploads.

To enable this, we create SubmittedContentClassifications for each AdditionalField. We then add a new Classification type to the UI

Additional Fields are submitted as key-value pairs. To specify these in ActionRules, users must write `<key>:<value>`. I will add this to the documentation.

I considered adding validation rules to ensure the value contains a colon but decided against as this may change in the future and users will likely have to look at the documentation anyway to use this feature

Test Plan
---------

1. Create a new ActionRule that looks for Submitted Content classified as `mobile_upload:true`
2. Submit a photo to HMA with Additional Fields `[{key: mobile_upload, value:true}, {key: another field, value : also true}]`
3. Watch webhook get called: https://webhook.site/#!/81bf390d-d2e3-4757-989d-9822a81d85dd/c28766d5-ddb5-47d8-9d40-423221ba47c8/1
4. Submit a photo to HMA with Additional Fields `[{key: mobile_upload, false}]`
5. Watch webhook NOT get called: https://webhook.site/#!/81bf390d-d2e3-4757-989d-9822a81d85dd/c28766d5-ddb5-47d8-9d40-423221ba47c8/1

Screen Shot 2021-08-03 at 6 12 35 PM

Screen Shot 2021-08-03 at 6 13 00 PM

Screen Shot 2021-08-03 at 6 12 22 PM

Screen Shot 2021-08-09 at 1 25 02 PM

Screen Shot 2021-08-09 at 1 25 16 PM

### CODE REVIEW

File: hasher-matcher-actioner/hmalib/common/classification_models.py
Comment (BarrettOlson) [inline hasher-matcher-actioner/hmalib/common/classification_models.py:54]: namings is hard but we should still try to have consistency here. It looks like all the other `ClassificationLabel` have that in their name not just `Classification`. Also, this is used to check against `additional_fields` yes? can we include the word 'fields` in the name at a minimum?

File: hasher-matcher-actioner/hmalib/lambdas/actions/action_evaluator.py
Comment (BarrettOlson) [inline hasher-matcher-actioner/hmalib/lambdas/actions/action_evaluator.py:148]: nit: the typing was not clear to me and I was confused by the `|`. I prefer we use `.union()` for future readers unless you feel strongly about it

File: hasher-matcher-actioner/webapp/src/components/settings/ActionRulesTableRow.tsx
Comment (BarrettOlson) [inline hasher-matcher-actioner/webapp/src/components/settings/ActionRulesTableRow.tsx:108]: Could you include a screenshot of this change if it is visbal on the setting UI? I have the same nit about naming as well. This one is worse imo because it is a Classification just a different kind right?

**Review by BarrettOlson — APPROVED** [review]
Stamping to unblock but please do not merge without first updating the name to be consistent with other `ClassificationLabel`s  and reflect the value used for actioning both in models and the UI. 
Strong preference for an overly verbose name than a non-specific one.

File: hasher-matcher-actioner/hmalib/lambdas/actions/action_evaluator.py
Comment (jeberl) [reply hasher-matcher-actioner/hmalib/lambdas/actions/action_evaluator.py:148]: Lol. Syntactic sugar isn't so sugary if no one knows what's happening.

Will change

File: hasher-matcher-actioner/webapp/src/components/settings/ActionRulesTableRow.tsx
Comment (jeberl) [reply hasher-matcher-actioner/webapp/src/components/settings/ActionRulesTableRow.tsx:108]: Added screenshot.

Discussed offline

File: hasher-matcher-actioner/hmalib/common/classification_models.py
Comment (BarrettOlson) [reply hasher-matcher-actioner/hmalib/common/classification_models.py]: @jeberl, why did we decided not to add `field` here? I was under the impression we were going to but it is possible I am misremembering.

**Pull request merged**
